protocol/src/main/resources/protocol.xml

```diff
@@ -144,6 +146,8 @@
     <field name="rejectionType" id="7" type="RejectionType"/>
     <!-- populated when RecordType is COMMAND_REJECTION, UTF-8-encoded String -->
     <data name="rejectionReason" id="8" type="varDataEncoding"/>
+    <!-- populated when RecordType is COMMAND -->
```

Comment:
This is only for authorization data. Other C8 components will also be able to get the same data through Identity.

Each RecordValue has a `tenantId` property, and that's how other components will determine the tenant ownership of the data. The authorization data in the metadata is only to be used internally in Zeebe to determine if we can process a command record.

Since we want to validate tenant access in the state, we would need to pass the authorization data when calling `StateWriter#appendFollowUpEvent(...)`.

We can discuss if we should then include the authorization data in the metadata of the actual event or not. My preference would be not to do it in the first iteration of multi-tenancy. We can add it in the future if it's needed.

WDYT?

Reply:
That makes sense. That means the[ issue I closed](https://github.com/camunda/zeebe/issues/13322) this morning is still relevant, so I'll reopen it. 

I do wonder if we should find a way to prevent the authorisation data from being exported at all. With this PR it's added to the index templated. We might not want to expose it at all.

Reply:
My proposal would be to skip setting the authorization data in the metadata when we create event records. I think it's fine if we export an empty property for now. In the future, we can decide to add it.

What do you think @remcowesterhoud ?

Reply:
As far as I can see we don't need this data anywhere for events. So I think that'd be fine.